index.js

```diff
@@ -99,30 +99,35 @@ UglifyWriter.prototype.processFile = function(inFile, outFile, relativePath, out
   var src = fs.readFileSync(inFile, 'utf-8');
   var mapName = path.basename(outFile).replace(/\.js$/,'') + '.map';
   var mapDir;
-  var origSourcesContent;
 
   if (this.sourceMapConfig.mapDir) {
     mapDir = path.join(outDir, this.sourceMapConfig.mapDir);
   } else {
     mapDir = path.dirname(path.join(outDir, relativePath));
   }
 
+  var sourceMapOptions = {
+    includeSources: this.options.sourceMapIncludeSources,
+    filename: inFile,
+    url: this.mapURL(mapName)
+  };
+
   var opts = {
-    fromString: true,
-    outSourceMap: this.mapURL(mapName),
-    enableSourcemaps: this.enableSourcemaps()
+    compress: this.options.compress,
+    mangle: this.options.mangle,
+    sourceMap: this.enableSourcemaps() ? sourceMapOptions : false
   };
 
-  if (opts.enableSourcemaps && srcURL.existsIn(src)) {
+  if (this.enableSourcemaps() && srcURL.existsIn(src)) {
     var url = srcURL.getFrom(src);
-    opts.inSourceMap = path.join(path.dirname(inFile), url);
-    origSourcesContent = JSON.parse(fs.readFileSync(opts.inSourceMap)).sourcesContent;
+    opts.sourceMap.content = JSON.parse(fs.readFileSync(path.join(path.dirname(inFile), url)));
   }
 
   try {
     var start = new Date();
     debug('[starting]: %s %dKB', relativePath, (src.length / 1000));
-    var result = UglifyJS.minify(src, merge(opts, this.options));
+    var result = UglifyJS.minify(src, opts, this.options);
+    if (result.error) throw result.error;
```

Comment:
UglifyJS 3 no longer throws an error for us, and instead expects us to throw it ourselves: This is documented at the end of the [API Reference section](https://github.com/mishoo/UglifyJS2#api-reference) of the readme.